crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -79,87 +92,70 @@ impl LogManagerActor {
     pub fn new(
         config: LogManagerConfig,
         plugin_config: LogPluginConfig,
-        mqtt_publisher: LoggingSender<MqttMessage>,
-        messages: SimpleMessageBox<LogInput, NoMessage>,
+        messages: SimpleMessageBox<LogInput, LogOutput>,
         upload_sender: DynSender<LogUploadRequest>,
     ) -> Self {
         Self {
             config,
             plugin_config,
             pending_operations: HashMap::new(),
-            mqtt_publisher,
             messages,
             upload_sender,
         }
     }
 
-    pub async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), ChannelError> {
-        if self.config.logfile_request_topic.accept(&message) {
-            match request_from_message(&message) {
-                Ok(Some(request)) => match request.status {
-                    CommandStatus::Init => {
-                        info!("Log request received: {request:?}");
-                        self.start_executing_logfile_request(&message.topic, request)
-                            .await?;
-                    }
-                    CommandStatus::Executing => {
-                        debug!("Executing log request: {request:?}");
-                        self.handle_logfile_request_operation(&message.topic, request)
-                            .await?;
-                    }
-                    CommandStatus::Scheduled
-                    | CommandStatus::Unknown
-                    | CommandStatus::Successful
-                    | CommandStatus::Failed { .. } => {}
-                },
-                Ok(None) => {}
-                Err(err) => {
-                    error!("Incorrect log request payload: {}", err);
-                }
+    pub async fn process_logfile_request(
+        &mut self,
+        request: LogUploadCmd,
+    ) -> Result<(), ChannelError> {
+        match request.status() {
+            CommandStatus::Init | CommandStatus::Scheduled => {
```

Comment:
A command has to be executed either on `init` or the `scheduled` state.

- Init is the trigger when the actor is directly exposed to MQTT (in the unit tests notably).
- When the actor is running behind the workflow actor, the trigger is the scheduled state.